### ISSUE TITLE
Flatten tool parameters need updating for BODS 0.4

### ISSUE THREAD

**Issue by kd-ods** [openownership/cove-bods#107]
**Bug description** 

At the moment, the flattened version of submitted BODS 0.4 data isn't quite right. If you run [this file](https://raw.githubusercontent.com/openownership/data-standard/refs/heads/example_drt_main/examples/tecido.json) through the tool, you'll see that there is no 'statementId' column on sheets 2 - 5 which would be needed to join the rows with the relevant rows on sheet 1. 

**Suggested fix**

The problem is that the parameters [passed to flatten-tool here](https://raw.githubusercontent.com/openownership/data-standard/refs/heads/example_drt_main/examples/tecido.json) are those for 0.3 and earlier. For 0.4, I think that id_name and root_id need to be 'statementId' and not 'statementID'.

**Comment by radix0000** [comment]
This specific problem should now be fixed in: http://bods-cove-addbods04.dokkubodsdev1.ods.mobi

Will close this issue once the inverse problem is also fixed.

**Comment by kd-ods** [comment]
Yes - let's ultimately test this @radix0000 by round-tripping some data from JSON -> xlsx -> JSON and checking that the checks give the same output along the way.

**Comment by radix0000** [comment]
Inverse has been added. Interestingly JSON -> xlsx -> JSON creates a load of extra errors even for BODS 0.3 data with the old code. Investigating.

**Comment by radix0000** [comment]
The obvious issues with this round trip in BODS 0.3 are the booleans (true/false in JSON) end up being strings of the Python boolean names ("True"/"False"), and also keys with a value of an empty list disappear along the way.
